### ISSUE TITLE
Add support for AWS credentials via IAM role

### ISSUE THREAD

**Issue by cfranklin11** [architect/architect#1311]
**Is your feature request related to a problem? Please describe.**
I'm trying to deploy from inside a CI agent, which is an EC2 instance with an IAM role, and I get "Error: Failed to deploy, missing or invalid AWS credentials or credentials file", because there isn't a credentials file or credentials env vars.

**Describe the solution you'd like**
I'd like for Architect to load AWS credentials from the IAM role when deploying.

**Describe alternatives you've considered**
I've started looking into generating temporary credentials from the IAM role and assigning these to the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars, but I'm far from an AWS CLI expert, and accomplishing this is not as simple as I had hoped.

**Additional context or notes**
Playing around a bit with loading IAM role credentials, I've found that calling `AWS.config.getCredentials()` fetches them, but is asynchronous. Looking at `@architect/utils/banner/init-aws.js`, the code is all synchronous, so this might not be a simple extension of current functionality. I'm happy to work on a PR, but would need some guidance on the preferred approach.

**Comment by ryanblock** [comment]
Sorry for the lag here @cfranklin11, have been heads down shipping Arc 10!

So to be completely honest, it's been so long since I've booted up an EC2 instance to run a Node.js app I really do not recall the correct approach to inherit the credentials from the EC2 IAM role. Per AWS's Node.js `aws-sdk` v2 docs, [we apparently don't have to do anything](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-iam.html), which seems incorrect.

It would be a pretty large lift to refactor how the banner initializes AWS credentials, so your suspicion is correct that that might be beyond reasonable scope for a PR – this code path touches critical paths of every single Architect workflow.

Just spitballing here, but have you considered writing a script to run `AWS.config.getCredentials()` prior to any Architect commands, and then populating AWS env vars with the results?

Open to other ideas as well!

**Comment by cfranklin11** [comment]
No worries. I managed to piece together a way to generate credentials from an EC2 IAM role from a bash script I found in another part of our codebase (using the `aws sts assume-role` command). Assigning the values of those credentials to the relevant AWS env vars before calling `arc deploy` was enough to get it to work.

>Per AWS's Node.js aws-sdk v2 docs, [we apparently don't have to do anything](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-iam.html), which seems incorrect.
My understanding of that, and other pages talking about how AWS SDK/CLI handle authorisation, is that the AWS object is able to use the EC2 IAM role for authorisation without needing to configure it specifically, but only auto-loads local credentials (e.g. from a file or env vars), because generating temporary credentials from the IAM role requires an asynchronous call (the `AWS.config.getCredentials()` mentioned above). I'm pretty sure I saw a section of the documentation saying this specifically, but I can't find the page again ☹️ 

Since this doesn't sound like a common use case, and the workaround is easy enough once you figure it out (if only the AWS documentation weren't so difficult to navigate), I'm happy to drop the suggested feature. One thing that might be helpful, though, would be to add an example of this scenario to the "Detailed AWS setup" section of the documentation. I'm happy to write something up if you think that would be useful.
